lib/output.js

```diff
@@ -521,7 +522,17 @@ function OutputStream(options) {
     // a function expression needs parens around it when it's provably
     // the first token to appear in a statement.
     PARENS(AST_Function, function(output){
-        return first_in_statement(output);
+        if (first_in_statement(output)) {
+            return true;
+        }
+
+        if (output.option('wrap_iife')) {
```

Comment:
`wrap_iife` is the wrong name for the option. It's wrapping all function expressions. Most of which are not IIFEs.